### DISCUSSION TITLE
dubbo proxyless 模式怎么指定通过本地 socket 连接控制面？

### DISCUSSION THREAD

**Discussion by Silocean** [apache/dubbo#13882] [General]
- [ ] I have searched the [issues](https://github.com/apache/dubbo/issues) of this repository and believe that this is not a duplicate.

![image](https://github.com/apache/dubbo/assets/6680721/719177fc-587f-46e1-a331-54e932f6395c)
在 3.2.7 源码中看到可以通过 url 中 use-agent 参数控制与控制面建立连接的方式：
use-agent=false 时会读取本地 agent 中的 bootstrap 文件，获取本地 socket 地址，通过该地址与控制面建立连接。

但是这个 url 是在哪里传入的？是 consumer 使用的注册中心地址配置吗？
![image](https://github.com/apache/dubbo/assets/6680721/25710423-06a8-4756-aec6-07a0f5e5e396)

我测试发现实际上默认use-agent=true，是否有配置可以指定 use-agent=false？应该在哪里配置？
![image](https://github.com/apache/dubbo/assets/6680721/c0204207-467a-4462-9387-94f0e2221fd5)

**Comment by sunheyi6** [comment]
> * [ ]  I have searched the [issues](https://github.com/apache/dubbo/issues) of this repository and believe that this is not a duplicate.
> 
> ![image](https://user-images.githubusercontent.com/6680721/279896679-719177fc-587f-46e1-a331-54e932f6395c.png) 在 3.2.7 源码中看到可以通过 url 中 use-agent 参数控制与控制面建立连接的方式： use-agent=false 时会读取本地 agent 中的 bootstrap 文件，获取本地 socket 地址，通过该地址与控制面建立连接。
> 
> 但是这个 url 是在哪里传入的？是 consumer 使用的注册中心地址配置吗？ ![image](https://user-images.githubusercontent.com/6680721/279897887-25710423-06a8-4756-aec6-07a0f5e5e396.png)
> 
> 我测试发现实际上默认use-agent=true，是否有配置可以指定 use-agent=false？应该在哪里配置？ ![image](https://user-images.githubusercontent.com/6680721/279898511-c0204207-467a-4462-9387-94f0e2221fd5.png)

url就是注册中心地址，可以通过配置文件配置use-agent=false

**Comment by Silocean** [comment]
![image](https://github.com/apache/dubbo/assets/6680721/4f7a8a9f-9545-487b-8315-17ed4dfa09a2)

@sunheyi6 我理解应该是这样配吧？但实际测试并没有生效🫤
